frontend/src/metabase/query_builder/actions/core/core.ts

```diff
@@ -330,22 +357,22 @@ export const revertToRevision = createThunkAction(
   },
 );
 
-async function reduxCreateQuestion(question, dispatch) {
+async function reduxCreateQuestion(question: Question, dispatch: Dispatch) {
   const action = await dispatch(Questions.actions.create(question.card()));
   return question.setCard(Questions.HACK_getObjectFromAction(action));
 }
 
 async function reduxUpdateQuestion(
-  question,
-  dispatch,
+  question: Question,
+  dispatch: Dispatch,
   { excludeDatasetQuery = false, excludeVisualisationSettings = false },
 ) {
   const fullCard = question.card();
 
   const keysToOmit = [
     excludeDatasetQuery ? "dataset_query" : null,
     excludeVisualisationSettings ? "visualization_settings" : null,
-  ].filter(Boolean);
+  ].filter(isNotNull);
```

Comment:
only typescript 5.5 is smart enough to figure out that we'll not have nulls in the array using `filter(Boolean)`